packages/compass-sidebar/src/index.ts

```diff
@@ -1,25 +1,63 @@
-import type { AppRegistry } from 'hadron-app-registry';
+import { registerHadronPlugin, type AppRegistry } from 'hadron-app-registry';
 import SidebarPlugin from './plugin';
-import SidebarStore from './stores';
+import { createSidebarStore } from './stores';
+import { mongoDBInstanceLocator } from '@mongodb-js/compass-app-stores/provider';
+import { dataServiceLocator } from 'mongodb-data-service/provider';
+import { type ConnectionInfo } from '@mongodb-js/connection-storage/renderer';
+import type { DataService } from 'mongodb-data-service';
+import type { MongoDBInstance } from 'mongodb-instance-model';
 
-/**
- * Activate all the components in the Sidebar package.
- * @param {Object} appRegistry - The Hadron appRegisrty to activate this plugin with.
- **/
-function activate(appRegistry: AppRegistry) {
-  appRegistry.registerComponent('Sidebar.Component', SidebarPlugin);
-  appRegistry.registerStore('Sidebar.Store', SidebarStore);
+function activate() {
```

Comment:
Can we remove this `activate` and `deactivate` now? Or are they supposed to be kept around for the time being?

Digging this new `registerHadronPlugin` usage !

Reply:
Right now, we are still loading all of these plugins into the PluginManager in the core compass package, and that interface expects these methods to be available. But that should be something we can get rid off after doing the rest of the conversion work :)